src/Catalyst.TestUtils/ContainerProvider.cs

```diff
@@ -0,0 +1,171 @@
+            ITestOutputHelper output = null)
+        {
+            _configFilesUsed = configFilesUsed;
+            _fileSystem = fileSystem;
+            _output = output;
+        }
+
+        protected string LogOutputTemplate { get; set; } =
+            "{Timestamp:HH:mm:ss} [{Level:u3}] ({ThreadId}) {Message} ({SourceContext}){NewLine}{Exception}";
+
+        protected LogEventLevel LogEventLevel { get; set; } = LogEventLevel.Verbose;
+
+        public IConfigurationRoot ConfigurationRoot
+        {
+            get
+            {
+                if (_configRoot != null)
+                {
+                    return _configRoot;
+                }
+
+                var configBuilder = new ConfigurationBuilder();
+                _configFilesUsed.ToList().ForEach(f => configBuilder.AddJsonFile(f));
+                
+                _configRoot = configBuilder.Build();
+                return _configRoot;
+            }
+        }
+
+        public IContainer Container => _container ?? (_container = ContainerBuilder.Build());
+
+        public void ConfigureContainerBuilder(bool writeLogsToTestOutput = false, bool writeLogsToFile = false, bool logDotNettyTraffic = false)
+        {
+            SocketPortHelper.AlterConfigurationToGetUniquePort(ConfigurationRoot);
+
+            var configurationModule = new ConfigurationModule(ConfigurationRoot);
+            ContainerBuilder.RegisterModule(configurationModule);
+            ContainerBuilder.RegisterModule(new CoreLibProvider());
+            ContainerBuilder.RegisterInstance(ConfigurationRoot).As<IConfigurationRoot>();
+
+            var repoFactory =
+                RepositoryFactory.BuildSharpRepositoryConfiguation(ConfigurationRoot.GetSection("CatalystNodeConfiguration:PersistenceConfiguration"));
+            ContainerBuilder.RegisterSharpRepository(repoFactory);
+
+            var passwordReader = new TestPasswordReader();
+            ContainerBuilder.RegisterInstance(passwordReader).As<IPasswordReader>();
+
+            var certificateStore = new TestCertificateStore();
+            ContainerBuilder.RegisterInstance(certificateStore).As<ICertificateStore>();
+            ContainerBuilder.RegisterInstance(_fileSystem).As<IFileSystem>();
+
+            var keyRegistry = TestKeyRegistry.MockKeyRegistry();
+            ContainerBuilder.RegisterInstance(keyRegistry).As<IKeyRegistry>();
+
+            ConfigureLogging(writeLogsToTestOutput, writeLogsToFile, logDotNettyTraffic);
+        }
+
+        private void ConfigureLogging(bool writeLogsToTestOutput, bool writeLogsToFile, bool logDotNettyTraffic = false)
+        {
+            var loggerConfiguration = new LoggerConfiguration()
+               .ReadFrom.Configuration(ConfigurationRoot).MinimumLevel.Verbose()
+               .Enrich.WithThreadId();
+
+            if (writeLogsToTestOutput)
+            {
+                if (_output == null)
+                {
+                    throw new NullReferenceException(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['System.NullReferenceException' should not be thrown by user code.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4318201)